components/schematics/ng-add/ng-add.ts

```diff
@@ -0,0 +1,305 @@
+
+/** Check workspace for files */
+function checkWorkspace(): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    try {
+      if (!tree.exists('package.json')) {
+        throw new Error('Cannot find package.json');
+      }
+
+      if (!tree.exists('angular.json')) {
+        throw new Error('Cannot find angular.json');
+      }
+    } catch (error) {
+      context.logger.error(error.message);
+      throw error;
+    }
+  };
+}
+
+/** Check for angular/barista-components in package.json */
+function checkForImportsInPackage(): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    const rules: Rule[] = [];
+    const packageJSON = readJsonInTree(tree, '/package.json');
+
+    if (
+      Object.keys(packageJSON.dependencies).includes(
+        '@dynatrace/angular-components',
+      )
+    ) {
+      delete packageJSON.dependencies['@dynatrace/angular-components'];
+
+      rules.push(renameExistingImports(['/package.json']));
+    }
+    packageJSON.dependencies['@dynatrace/barista-components'] = '5.0.0';
+
+    tree.overwrite('/package.json', JSON.stringify(packageJSON, null, 2));
+
+    return chain(rules)(tree, context);
+  };
+}
+
+/** Check filesystem for imports of dynatrace/angular-components and rename then to barista-components */
+function checkForExistingAngularComponents(): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    const rules: Rule[] = [];
+
+    // check if we have existing components library installed
+    if (
+      checkPackageJsonImports(
+        tree,
+        'package.json',
+        '@dynatrace/barista-components',
+      )
+    ) {
+      // if we have an existing library we have to rename the imports
+      const files = getFilesToCheck('**/*.ts');
+
+      rules.push(renameExistingImports(files));
+    }
+    return chain(rules)(tree, context);
+  };
+}
+
+function checkForPeerDependencies(): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    const rules: Rule[] = [];
+
+    const packages = getFilesToCheck('**/package.json');
+
+    for (const pkg of packages) {
+      const packageJSON = readJsonInTree(tree, pkg); // use 'pkg'
+      // install peerDependencies
+      if (packageJSON.peerDependencies) {
+        packageJSON.peerDependencies['@dynatrace/barista-icons'] =
+          '>= 3.0.0 < 4';
+        packageJSON.peerDependencies['@dynatrace/barista-fonts'] =
+          '>= 1.0.0 < 2';
+        packageJSON.peerDependencies['@types/highcharts'] = '^5.0.23';
+        packageJSON.peerDependencies['d3-scale'] = '^3.0.0';
+        packageJSON.peerDependencies['d3-shape'] = '^1.3.5';
+        packageJSON.peerDependencies['highcharts'] = '^6.0.7';
+        tree.overwrite(pkg, JSON.stringify(packageJSON, null, 2));
+      }
+    }
+    return chain(rules)(tree, context);
+  };
+}
+
+function installAnimationsModule(options: Schema): Rule {
+  return (tree: Tree, _: SchematicContext) => {
```

Comment:
Can you add some comments inside the function to explain the process a little bit.